cql3/statements/create_view_statement.cc

```diff
@@ -299,6 +301,35 @@ view_ptr create_view_statement::prepare_view(data_dictionary::database db) const
                 column_family(), column_names));
     }
 
+    // IS NOT NULL restrictions are handled separately from other restrictions.
+    // They need a separate check as they won't be included in non_pk_restrictions.
+    std::vector<std::string_view> invalid_not_null_column_names;
+    for (const column_definition* not_null_cdef : restrictions->get_not_null_columns()) {
+        if (!target_primary_keys.contains(not_null_cdef)) {
+            invalid_not_null_column_names.push_back(not_null_cdef->name_as_text());
+        }
+    }
+
+    if (!invalid_not_null_column_names.empty() &&
+        db.get_config().strict_is_not_null_in_views() == db::tri_mode_restriction_t::mode::TRUE) {
+        throw exceptions::invalid_request_exception(
+            fmt::format("The IS NOT NULL restriction is allowed only columns which are part of the view's primary key,"
+                        " found columns: {}. The flag strict_is_not_null_in_views can be used to turn this error "
+                        "into a warning, or to silence it. (true - error, warn - warning, false - silent)",
+                        fmt::join(invalid_not_null_column_names, ", ")));
+    }
+
+    if (!invalid_not_null_column_names.empty() &&
+        db.get_config().strict_is_not_null_in_views() == db::tri_mode_restriction_t::mode::WARN) {
+        sstring warning_text = fmt::format(
+            "The IS NOT NULL restriction is allowed only columns which are part of the view's primary key,"
+            " found columns: {}. Restrictions on these columns will be silently ignored. "
```

Comment:
nitpick: "silently" isn't exactly the right word to use in a warning message ;-)

Reply:
Well, they "will be" ignored, but aren't at the moment :P

I could fix it, but I see that you already merged the PR, so I'll leave it as is.